File: ClinicSync/Core/DTO/CreateAppointmentRequest.cs
using Core.Entities;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Core.DTO
{
    public class CreateAppointmentRequest
    {
        [Required]
        public Guid DoctorId { get; set; }

        [Required]
        public DateTime AppointmentDate { get; set; }

        [Required]
        public TimeSpan StartTime { get; set; }

        [MaxLength(500)]
        public string? ReasonForVisit { get; set; }
    }

    public class UpdateAppointmentRequest
    {
        [MaxLength(500)]
        public string? ReasonForVisit { get; set; }

        [MaxLength(1000)]
        public string? Notes { get; set; }
    }

    public class CancelAppointmentRequest
    {
        [Required]
        [MaxLength(500)]
        public string CancellationReason { get; set; } = string.Empty;
    }

    public class UpdateAppointmentStatusRequest
    {
        [Required]
        public AppointmentStatus Status { get; set; }

        [MaxLength(1000)]
        public string? Notes { get; set; }
    }

    public class DoctorAvailabilityRequest
    {
        [Required]
        public Guid DoctorId { get; set; }

        [Required]
        public DateTime Date { get; set; }
    }
}

 

File: ClinicSync-UI/src/app/shared/models/CreateAppointmentRequest .ts
export interface CreateAppointmentRequest {
  doctorId: string;
  appointmentDate: string; // ISO string
  startTime: string; // HH:mm format
  reasonForVisit?: string;
}

export interface UpdateAppointmentRequest {
  reasonForVisit?: string;
  notes?: string;
}

export interface CancelAppointmentRequest {
  cancellationReason: string;
}

export interface UpdateAppointmentStatusRequest {
  status: AppointmentStatus;
  notes?: string;
}

export interface DoctorAvailabilityRequest {
  doctorId: string;
  date: string; // ISO string
}

export interface AppointmentResponse {
  id: string;
  referenceNumber: string;
  appointmentDate: string;
  startTime: string;
  endTime: string;
  status: string;
  reasonForVisit?: string;
  notes?: string;
  createdAt: string;
  cancelledAt?: string;
  cancellationReason?: string;
  doctor: DoctorInfo;
  patient?: PatientInfo;
}

export interface DoctorInfo {
  id: string;
  fullName: string;
  specialty: string;
  consultationFee: number;
  profilePictureUrl?: string;
}

export interface PatientInfo {
  id: string;
  fullName: string;
  email: string;
  phoneNumber?: string;
}

export interface TimeSlotResponse {
  startTime: string;
  endTime: string;
  isAvailable: boolean;
}

export interface AvailabilityResponse {
  success: boolean;
  message: string;
  timeSlots?: TimeSlotResponse[];
}

export interface PaginatedAppointmentsResponse {
  success: boolean;
  message: string;
  data?: AppointmentResponse[];
  pageNumber: number;
  pageSize: number;
  totalPages: number;
  totalCount: number;
  hasPrevious: boolean;
  hasNext: boolean;
}

export enum AppointmentStatus {
  Confirmed = 'Confirmed',
  Completed = 'Completed',
  Cancelled = 'Cancelled',
  NoShow = 'NoShow'
}